scripts/ButterSpreadContainer.gd

```diff
@@ -0,0 +1,23 @@
+extends Node2D
+
+# "0" is multi node, "1" is single node (newer version) 
+var butter_version = 0
+
+var butter_scene = load("res://scenes/ButterSpreadParticle.tscn")
+var butter_scene_v2 = load("res://scenes/ButterSpreadParticle_v2.tscn")
+var butter_instance_v2 = butter_scene_v2.instance()
+
+func _ready():
+	butter_instance_v2.emitting = false
```

Comment:
Set in the option of ButterSpreadParticle_v2  `OneShot = true`, then you do not have to change emitting every time to false in _ready function; Godot will do that for you :))

Reply:
> I don't agree. Using <one_shot> property would not fit in here because new particles wouldn't be drawn unless I restart the node or (as we were doing earlier) duplicate it indefinitely. the property allows my to control creating new particles from one node. I do think this is better

Ok, I don't know why but I thought that _ready is called at all times. So I don't know how it works. I thought that you have to use one_shot to emit particles one time after moving and stop emitting after that. So my question is in that case why does it not emit particles all the time? xd
And I've checked my version works too. I set one_shot to true and in _on_GamePlay_Moved i change emitting to true ;))

Reply:
yes, _ready() is called only once - and yes, you're right about one_shot this time.
At first I thought that signal was working like a state machine and only when player has just started moving I would envoke "emitting" once (and it would cycle itself) and only when player has just stopped moving I would set "emitting" to false (not yet implemented in code). It seemed to me that there would be less calls, but eventually  there aren't as I'm passing "Player's" actual position through this signal too

Reply:
maybe we can merge this now and make changes to this afterwards? or should I commit new changes here and compare to the latest branch? whats youtr proposition

Reply:
or commit to #7  ?

Reply:
Ok so I will merge this and we will think about it later :))